compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -411,6 +411,18 @@ impl SirFuncCx<'tcx> {
                 self.lower_binop(bx, bb, dest_ty, *op, opnd1, opnd2, true)
             }
             mir::Rvalue::Cast(mir::CastKind::Misc, op, ty) => self.lower_cast_misc(bx, bb, op, ty),
+            mir::Rvalue::Len(p) => {
+                // Yorick's register allocator will always put a fat pointer on the stack, so
```

Comment:
`Rvalue::Len` on arrays is also valid: https://github.com/bjorn3/rustc_codegen_cranelift/blob/54ba238a6d4d1c2106848ffeb44b6932e95d27fa/src/base.rs#L916-L933 Looking at https://github.com/bjorn3/rustc_codegen_cranelift/commit/f4e544894abc6424d70ec27c196ef4d722dd9379 it seems to be done in the drop glue for `[T; N]`.

Reply:
If I understand properly, in the case of an array, we can statically insert a constant length? I think I'd like to do that as a separate PR, but here we should at least insert an "unimplemented" statement for arrays.

Sounds like a sensible plan?

Reply:
So I think we're just waiting on this one?

Reply:
I've implemented most of it, but we are hitting an `unimplemented!` now. Bear with me.

Reply:
OK :)

Reply:
Fixed in 2fb17a4.